sublime_jedi/helper.py

```diff
@@ -33,28 +49,129 @@ def docstring_panel(view, docstring):
+        location=location,
+        max_width=800,
+        md=True,
+        css=css,
+        wrapper_class='jedi',
+        flags=sublime.HIDE_ON_MOUSE_MOVE_AWAY)
+
+
+def markdown_html_builder(view, docstring):
+    doclines = docstring.split('\n')
+    signature = signature_builder(doclines[0])
+    if signature:
+        # highlight signature
+        content = '```python\n{0}\n```\n'.format(signature)
+        # merge the rest of the docstring beginning with 3rd line
+        # skip leading and tailing empty lines
+        docstring = '\n'.join(doclines[1:]).strip()
+        content += html.escape(docstring, quote=False)
     else:
-        sublime.status_message('Jedi: No results!')
+        # docstring does not contain signature
+        content = html.escape(docstring, quote=False)
+
+    # preserve empty lines
+    content = content.replace('\n\n', '\n\u00A0\n')
+    # preserve whitespace
+    content = content.replace('  ', '\u00A0\u00A0')
+    # convert markdown to html
+    content = mdpopups.md2html(view, content)
+    # highlight headlines ( Google Python Style Guide )
+    keywords = (
+        'Args:', 'Arguments:', 'Attributes:', 'Example:', 'Examples:', 'Note:',
+        'Raises:', 'Returns:', 'Yields:')
+    for keyword in keywords:
+        content = content.replace(
+            keyword + '<br />', '<h6>' + keyword + '</h6>')
+    return content
+
+
+def docstring_tooltip_simple(view, docstring, location):
+    """Show docstring in popup using simple renderer.
+
+    :param view (sublime.View): current active view
+    :param docstring (basestring): python __doc__ string
+    :param location (int): The text point where to create the popup
+    """
+    content = simple_html_builder(docstring)
+    return view.show_popup(content, location=location, max_width=512)
 
 
 def simple_html_builder(docstring):
-    docstring = docstring.split('\n')
+    docstring = html.escape(docstring, quote=False).split('\n')
     docstring[0] = '<b>' + docstring[0] + '</b>'
-    html = '<body><p style="font-family: sans-serif; font-family: sans-serif;">{0}</p></body>'.format(
+    content = '<body><p style="font-family: sans-serif;">{0}</p></body>'.format(
        '<br />'.join(docstring)
     )
-    return html
+    return content
+
+
+def signature_builder(string):
+    """Parse string and prepend def/class keyword for valid signature.
+
+    :param string: The string parse and check whether it is a signature.
+
+    :returns: None string or the prefixed signature
+    """
+    pattern = '^([\w\.]+\.)?(\w+)\('
+    match = re.match(pattern, string)
+    if not match:
+        return None
+
+    path, func = match.groups()
+    # lower case built-in types
+    types = ('dict', 'int', 'list', 'tuple', 'str', 'set', 'frozenset')
+    if any(func.startswith(s) for s in types):
+        prefix = ''
+    else:
+        func = func.lstrip('_')
+        prefix = 'class ' if func[0].isupper() else 'def '
```

Comment:
Currently we receive docstring it self. 
But I think we should change it to build more valuable tooltips.
Jedi API allows to do this.